### PR TITLE
refactor: extract MarkdownEditor component

### PR DESCRIPTION
**Pull request by Sekhmet**
### Summary

Closes: https://github.com/snapshot-labs/sx-ui/issues/788

### How to test

1. Go to proposal editor.
2. Editor still works.

### CODE REVIEW

File: src/components/MarkdownEditor.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+});
+
+const previewEnabled = ref(false);
+</script>
+
+<template>
+  <div class="flex">
+    <Link :is-active="!previewEnabled" text="Write" class="pr-3" @click="previewEnabled = false" />
+    <Link :is-active="previewEnabled" text="Preview" @click="previewEnabled = true" />
+  </div>
```
Comment (bonustrack) [inline src/components/MarkdownEditor.vue:24]: Can we just move this out of the component?

Comment (Sekhmet) [reply src/components/MarkdownEditor.vue:24]: Just elements to switch between two modes (and passing previewEnabled as prop to display preview within this component), or entire preview functionality (so consumer of this component will need its own preview component)?

Comment (bonustrack) [reply src/components/MarkdownEditor.vue:24]: I mean the entire preview functionality

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**